File: test-suite-kotlin/src/test/kotlin/io/micronaut/docs/config/property/EngineSpec.kt
package io.micronaut.docs.config.property

import io.kotlintest.shouldBe
import io.kotlintest.specs.StringSpec
import io.micronaut.context.ApplicationContext
import org.junit.Test

import java.util.LinkedHashMap

import org.junit.Assert.assertEquals

class EngineSpec : StringSpec({

    "test start vehicle with configuration" {
        val ctx = ApplicationContext.run(mapOf("my.engine.cylinders" to "8", "my.engine.manufacturer" to "Honda"))

        val engine = ctx.getBean(Engine::class.java)

        engine.manufacturer shouldBe "Honda"
        engine.cylinders() shouldBe 8

        ctx.close()
    }
})

File: test-suite-python/src/test/python/micronaut/docs/factories/EngineSpec.py
from org.junit.jupiter.api import Test
from micronaut.test.extensions.junit5.annotation import MicronautTest
from micronaut.context import ApplicationContext
from micronaut.context.exceptions import NoSuchBeanException
from jakarta.inject import Inject
from typing import Annotated
import java

# tag::class[]
@MicronautTest
class EngineSpec:
    context : Annotated[ApplicationContext, Inject] = None

    @Test
    def test_engine(self):
        # tag::start[]
        Engine = java.type("micronaut.docs.factories.Engine")
        assert self.context.getBean(Engine) is not None
        # end::start[]

# end::class[]
